wemake_python_styleguide/visitors/ast/operators.py

```diff
@@ -223,10 +223,29 @@ def _check_string_concat(
 class WalrusVisitor(base.BaseNodeVisitor):
     """We use this visitor to find walrus operators and ban them."""
 
+    _available_parents: ClassVar[AnyNodes] = (
+        ast.ListComp,
+        ast.SetComp,
+        ast.DictComp,
+        ast.GeneratorExp,
+    )
+
     def visit_NamedExpr(
         self,
         node: ast.NamedExpr,
     ) -> None:
-        """Disallows walrus ``:=`` operator."""
-        self.add_violation(consistency.WalrusViolation(node))
+        """Disallows walrus ``:=`` operator outside comprehensions."""
+        closest = walk.get_closest_parent(
```

Comment:
Please, create a method for that. Let's call it `_check_walrus_in_comprehension`